fhir-server/src/main/java/org/linuxforhealth/fhir/server/util/FHIRRestHelper.java

```diff
@@ -2466,6 +2500,10 @@ private Bundle addLinks(FHIRPagingContext context, Bundle responseBundle, String
         if (selfUri == null) {
             selfUri = requestUri;
         }
+        // add the resource Id of the first resource of the search result as a query parameter to self uri
+        selfUri = addParameterToUrl(selfUri, "firstId", firstId);
```

Comment:
maybe use the new SearchConstants here?
if so, we'd need to update `addParameterToUrl` to avoid prepending the `_` since the constants already have that.

Reply:
This has been fixed as suggested.